### PR TITLE
Validate the `req.params._id` and `_id` property to ensure it's a valid ObjectId

### PR DESCRIPTION
**Pull request by IgnatG**
### Changelog:
- added a validation for `req.params.id` properties in central-api to check whether an `_id` is a valid Object Id. If it's not, then the server crashes because it tries to query the DB with an invalid _id

### CODE REVIEW

File: dtfs-central-api/src/v1/controllers/portal/deal/add-deal-comment.controller.js [hunk 1]
```diff
@@ -5,41 +5,47 @@ const { findOneDeal } = require('./get-deal.controller');
...
-    { _id: ObjectId(_id) },
-    {
-      $push: {
-        [commentType]: {
-          $each: [{
-            ...comment,
-            timestamp: Date.now(),
-          }],
-          $position: 0,
+  if (ObjectId.isValid(_id)) {
```
Comment (ttbarnes) [inline dtfs-central-api/src/v1/controllers/portal/deal/add-deal-comment.controller.js:8]: Do we need this condition here? It's done in the POST function below

Comment (ttbarnes) [reply dtfs-central-api/src/v1/controllers/portal/deal/add-deal-comment.controller.js:8]: Noticed this in all the functions - I imagine it's fail safe with some of the functions that will be called without the "crud endpoint"? Happy to approve.

**Review by ttbarnes — APPROVED** [review]

File: dtfs-central-api/src/v1/controllers/portal/deal/add-deal-comment.controller.js (hunk 1 above)
Comment (IgnatG) [reply dtfs-central-api/src/v1/controllers/portal/deal/add-deal-comment.controller.js:8]: exactly. it should be handled as soon as the endpoint is called. but sometimes we import those functions in other files and it skips that validation from "crud endpoint". So yes, it's a fail-safe :)

Comment (ttbarnes) [reply dtfs-central-api/src/v1/controllers/portal/deal/add-deal-comment.controller.js:8]: Brilliant 😄

**Pull request merged**